Modify rules S1871 and S3923: Update description in accordance with the new understanding (#3752)
SonarSource/rspec: rules/S1871/cfamily/exceptions.adoc

=== Exceptions

> Branches in an `if` or `switch` chain that have a _simple_ body are ignored. A body is considered simple if it contains a single line of code, possibly followed by a `break` in case of `switch` chains.

Reviewer: Nit: "switch chain" is not a thing AFAIK.

```suggestion
Branches in `if` or `switch` statements that have a _simple_ body are ignored. A body is considered simple if it contains a single line of code, possibly followed by a `break` in the case of ``switch``es.
```
